### ISSUE TITLE
Disabled Lights Are Included in Lightmap

### ISSUE THREAD

**Issue by sandsalamand** [godotengine/godot#82834]
### Godot version

4.1.2 stable mono

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2070 SUPER (NVIDIA; 30.0.15.1123) - Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 Threads)

### Issue description

Invisible lights (tested with OmniLight3D and SpotLight3D) are included in the baked lightmap.

Expected: disabled/invisible lights are not included in the lightmap.

### Steps to reproduce

1. Place an OmniLight3D or SpotLight3D in the scene.
2. Nest it below a LightMapGI
3. Place a MeshInstance3D in the scene (in my case, saved as a scene, but not sure if that matters).
4. Bake. Lightmap settings don't seem to change anything, but my last test used Low + 3 Bounces + Directional ON + Interior OFF + Use Denoiser ON.

### Minimal reproduction project

N/A

**Comment by jsjtxietian** [comment]
Hi, by invisible lights do you mean toggle visibility at scene hierachy or change the bake mode to disabled? If the first situation, the doc said "Keep in mind hiding a light will have no effect for baking, so this must be used instead of hiding the Light node." https://docs.godotengine.org/en/latest/tutorials/3d/global_illumination/using_lightmap_gi.html

However, I do think invisible light in the scene hierarchy should not contribute to lightmap baking.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/issues/56611.

This may be done by design, so that you can maximize performance by having lights that only affect baked surfaces.

**Comment by jsjtxietian** [comment]
Well I would slightly argue that it's not intuitive that invisible light is also contributing to lightmap baking. Other than this it's fine.

**Comment by sandsalamand** [comment]
@Calinou So from reading that PR, are you saying that Static Baked lights would be completely invisible in the editor until baked? And thus, they would solve the use case of wanting lights that are invisible in the editor but included in the lightmap? 

If so, I agree. If that option was available, then we could safely change it so that invisible dynamic lights are not included in the lightmap.

**Comment by sandsalamand** [comment]
@jsjtxietian It's not essential right now for my project, but it's definitely unintuitive behavior that should (IMO) be changed.

I read that documentation page, but I didn't notice that part. Maybe it's because of the use of the word "hidden" instead of "invisible". Also, if we're keeping the current behavior, it should probably be in a warning box to call more attention to it.

**Comment by Calinou** [comment]
> @Calinou So from reading that issue, are you saying that Static Baked lights would be completely invisible in the editor until baked? And thus, they would solve the use case of wanting lights that are invisible in the editor but included in the lightmap?

This would be an additional bake mode for the reasons I described in https://github.com/godotengine/godot/issues/73289#issuecomment-1430259719. The existing Static bake mode would not be affected by this change, as it still has some use cases (such as more precise direct lighting for direct objects, and the ability for them to *receive* real-time shadows from baked objects).

**Comment by sandsalamand** [comment]
Yes, I understand. So, should we add the fix which @jsjtxietian mentioned to your fork?

**Comment by Calinou** [comment]
> Yes, I understand. So, should we add the fix which @jsjtxietian mentioned to your fork?

I would prefer we make the changes separately, as it has important implications. My branch is still far from being finished (and I don't know if I'll be able to complete it) anyway.

**Comment by atirut-w** [comment]
This is very similar to a problem I encountered where editor-only lights are included in lightmaps. Could it be the same issue as this?

**Comment by Calinou** [comment]
> This is very similar to a problem I encountered where editor-only lights are included in lightmaps. Could it be the same issue as this?

This can likely be the fixed the same way, but unlike hidden lights, I consider including editor-only lights in the bake a bug for sure.

**Comment by sandsalamand** [comment]
@atirut-w As you can see in the code @jsjtxietian posted, the engine simply doesn't check whether the light is editor-only or disabled. So yes, we could fix both problems at once.

**Comment by reptofrog** [comment]
I have attached a reproduction project, I hope it could be useful.

[Lightmap Bug.zip](https://github.com/godotengine/godot/files/14705537/Lightmap.Bug.zip)

**Comment by sandsalamand** [comment]
@reptofrog Are you a bot?

**Comment by Calinou** [comment]
> @reptofrog Are you a bot?

No, I can attest they are a real person 🙂

**Comment by sandsalamand** [comment]
> > @reptofrog Are you a bot?
> 
> No, I can attest they are a real person 🙂 

Ok, it just seemed weird because they randomly posted a picture and a .zip file without any context.

**Comment by reptofrog** [comment]
@sandsalamand 
It's a reproduction project that demonstrates how disabled lights are included in the lightmap 😅

**Comment by Zireael07** [comment]
@sandsalamand The post literally says "a reproduction project" above the picture and zip

**Comment by sandsalamand** [comment]
@Zireael07 Right, but Calinou expressed doubt that this behavior is actually a bug, stating that it might be intentional that invisible lights are included. So it seemed weird for reptofrog to upload a file called "Lightmap Bug".
